applications/matter_bridge/src/bridged_device_types/ble_environmental_data_provider.cpp

```diff
@@ -0,0 +1,269 @@
+
+	if (CheckSubscriptionParameters(&mGattTemperatureSubscribeParams)) {
+		int err = bt_gatt_subscribe(mDevice->mConn, &mGattTemperatureSubscribeParams);
+		if (err) {
+			LOG_ERR("Subscribe to temperature characteristic failed with error %d", err);
+		}
+	} else {
+		LOG_ERR("Invalid temperature subscription parameters provided");
+	}
+
+	if (CheckSubscriptionParameters(&mGattHumiditySubscribeParams)) {
+		int err = bt_gatt_subscribe(mDevice->mConn, &mGattHumiditySubscribeParams);
+		if (err) {
+			LOG_ERR("Subscribe to humidity characteristic failed with error %d", err);
+		}
+	} else {
+		LOG_INF("Invalid humidity subscription parameters provided, starting emulated subscription");
+		/* The fallback is based on polling for the current GATT humidity value */
+		sHumidityReadParams.func = BleEnvironmentalDataProvider::HumidityGATTReadCallback;
+		sHumidityReadParams.handle_count = 0;
+		sHumidityReadParams.by_uuid.start_handle = BT_ATT_FIRST_ATTRIBUTE_HANDLE;
+		sHumidityReadParams.by_uuid.end_handle = BT_ATT_LAST_ATTRIBUTE_HANDLE;
+		sHumidityReadParams.by_uuid.uuid = sUuidHumidity;
+		StartHumidityTimer();
+	}
+}
+
+void BleEnvironmentalDataProvider::Unsubscribe()
+{
+	VerifyOrReturn(mDevice && mDevice->mConn, LOG_ERR("Invalid connection object"));
+
+	int err = bt_gatt_unsubscribe(mDevice->mConn, &mGattTemperatureSubscribeParams);
+	if (err) {
+		LOG_INF("Cannot unsubscribe from temperature characteristic (error %d)", err);
+	}
+
+	err = bt_gatt_unsubscribe(mDevice->mConn, &mGattHumiditySubscribeParams);
+	if (err) {
+		LOG_INF("Cannot unsubscribe from humidity characteristic (error %d)", err);
+	}
+	StopHumidityTimer();
+}
+
+bool BleEnvironmentalDataProvider::CheckSubscriptionParameters(bt_gatt_subscribe_params *params)
+{
+	/* If any of these is not met, the bt_gatt_subscribe() generates an assert at runtime */
+	VerifyOrReturnValue(params && params->notify, false);
+	VerifyOrReturnValue(params->value, false);
+	VerifyOrReturnValue(params->ccc_handle, false);
+
+	return true;
+}
+
+void BleEnvironmentalDataProvider::NotifyTemperatureAttributeChange(intptr_t context)
+{
+	BleEnvironmentalDataProvider *provider = reinterpret_cast<BleEnvironmentalDataProvider *>(context);
+
+	provider->NotifyUpdateState(Clusters::TemperatureMeasurement::Id,
+				    Clusters::TemperatureMeasurement::Attributes::MeasuredValue::Id,
+				    &provider->mTemperatureValue, sizeof(provider->mTemperatureValue));
+}
+
+void BleEnvironmentalDataProvider::NotifyHumidityAttributeChange(intptr_t context)
+{
+	BleEnvironmentalDataProvider *provider = reinterpret_cast<BleEnvironmentalDataProvider *>(context);
+
+	provider->NotifyUpdateState(Clusters::RelativeHumidityMeasurement::Id,
+				    Clusters::RelativeHumidityMeasurement::Attributes::MeasuredValue::Id,
+				    &provider->mHumidityValue, sizeof(provider->mHumidityValue));
+}
+
+void BleEnvironmentalDataProvider::StartHumidityTimer()
+{
+	static constexpr uint32_t kMeasurementsIntervalMs{ 3000 };
```

Comment:
What about making this value a constexpr/kconfig somewhere else than in the method body?